### ISSUE TITLE
Suggestion: Fueled Soul Lantern

### ISSUE THREAD

**Issue by juicebin** [WolfieWaffle/Hardcore-Torches-Forge#27]
I feel that the soul lantern should also be a fueled lantern!

The items could be `Unlit Soul Lantern` and `Lit Soul Lantern`, the unlit variant having a patch of soul sand in the bottom.

Perhaps it doesn't need to have as much oil as a regular lantern, giving it a reason to use? Then to balance, it could emit a bit less light.

**Comment by WolfieWaffle** [comment]
Soul torches and lanterns will probably be the next thing I do, and I'm going to use a suggestion by someone else to use the user's XP as the fuel source.

Though if I remember, soul fire is infinite, and fire is infinite with netherrack, so once you get to the nether you're gonna basically have free light anyways. But, maybe it will be useful for modpacks.

**Comment by juicebin** [comment]
Yes, I'd say it's useful for modpacks! In my modpack, soul lanterns are naturally generated in a specific type of village and certain dungeons (am working on disabling it for now). So players can obtain them prior to the Nether 😅

Perhaps it could even be configurable? 🤷

**Comment by WolfieWaffle** [comment]
I think I forgot to mention this but I did eventually add these. Though, they aren't super developed as of yet. You craft them with soul sand surrounded by iron nuggets, and you have to right click them with a candle to transfer XP into them as fuel. Working on in-inventory transfer as well as binding yourself to lanterns to constantly draw XP or something.
